src/main/java/com/aws/greengrass/mqtt/bridge/clients/IoTCoreClient.java

```diff
@@ -44,26 +44,26 @@ public class IoTCoreClient implements MessageClient {
     private Set<String> subscribedIotCoreTopics = ConcurrentHashMap.newKeySet();
     @Getter(AccessLevel.PROTECTED)
     private Set<String> toSubscribeIotCoreTopics = new HashSet<>();
-
-    private Consumer<Message> messageHandler;
+    private volatile Consumer<Message> messageHandler;
```

Comment:
it's set every time `updateSubscriptions` is called, but yeah it doesn't matter all that much right now since the lambda content doesn't change `message -> handleMessage(clientType, message)`, but it's technically a new reference each time